### PR TITLE
add retry feature on Libcloud error on upload/deletions

### PR DESCRIPTION
**Pull request by rphillips**

### CODE REVIEW

File: file_syncer/syncer.py [hunk 1]
```diff
@@ -212,6 +212,10 @@ def _remove_object(self, item):
 
         try:
             driver.delete_object(obj=obj)
+        except LibcloudError, e:
```
Comment (Kami) [inline file_syncer/syncer.py:230]: Should log something and limit number of retries.

**Comment by rphillips** [comment]
@Kami Updated the PR with limits

File: file_syncer/syncer.py [hunk 2]
```diff
@@ -191,6 +194,18 @@ def _generate_manifest(self, remote_files):
 
         return manifest
 
+    def _register_retry(self, name):
+        if not self._retries.get(name):
```
Comment (Kami) [inline file_syncer/syncer.py]: More Pythonic way would be `if name not in self._retries` or using `defaultdict`

File: file_syncer/syncer.py [hunk 3]
```diff
@@ -191,6 +194,18 @@ def _generate_manifest(self, remote_files):
...
+            self._retries[name] = 1
+        else:
+            self._retries[name] = self._retries[name] + 1
+
+        return self._retries[name] <= self._retry_limit
+
+    def _clear_retry(self, name):
+        if self._retries.get(name):
```
Comment (Kami) [inline file_syncer/syncer.py]: Same here (`name in self._retries`).

File: file_syncer/syncer.py (hunk 2 above)
Comment (rphillips) [inline file_syncer/syncer.py]: fixed

File: file_syncer/syncer.py (hunk 3 above)
Comment (rphillips) [inline file_syncer/syncer.py]: fixed.

File: file_syncer/syncer.py [hunk 4]
```diff
@@ -191,6 +194,18 @@ def _generate_manifest(self, remote_files):
 
         return manifest
 
+    def _register_retry(self, name):
```
Comment (Kami) [inline file_syncer/syncer.py]: I would also call this function `_should_retry`, otherwise LGTM.

Comment (rphillips) [inline file_syncer/syncer.py]: fixed.

**Comment by Kami** [comment]
Merged, thanks.

**Comment by rphillips** [comment]
When you get a chance, can you do a pypi release?

**Comment by Kami** [comment]
New version has been published to PyPi - http://pypi.python.org/pypi/file_syncer/

**Pull request merged**